### PR TITLE
chore: replace harness with scenario

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

Replace harness with scenario

## Rationale

The scenario framework enforces our current approach with:
- Arrange (state in)
- Act (juju event)
- Assert (state out)

This makes tests easier to read, to write, and to parameterise .

## Reference
- https://github.com/canonical/ops-scenario

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/unit/fixtures.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+class MockNetwork:
+    def __init__(self, bind_address: str, ingress_address: str):
+        self.bind_address = bind_address
+        self.ingress_address = ingress_address
+
+
+class MockBinding:
+    def __init__(self, bind_address: str, ingress_address: str):
+        self.network = MockNetwork(bind_address=bind_address, ingress_address=ingress_address)
```
Comment (kayra1) [inline tests/unit/fixtures.py:118]: Scenario should already have built-in Network mocking support like [so](https://github.com/canonical/gocert-k8s-operator/blob/4ab6771ca20aa38fa6b80639f97d1053e5bc41dc/tests/unit/test_charm.py#L1457). In the case of vault, the relation string should be 'vault-peers' instead.

File: tests/unit/test_charm_authorize_action.py
Comment (kayra1) [inline tests/unit/test_charm_authorize_action.py:1]: I really like splitting tests by the handler

**Review by kayra1 — CHANGES_REQUESTED** [review]
awesome. just one comment.

File: tests/unit/test_charm_authorize_action.py
Comment (gruyaume) [reply tests/unit/test_charm_authorize_action.py:1]: Me too!

**Review by kayra1 — APPROVED** [review]

**Pull request merged**